### ISSUE TITLE
How to check whether sound is still playing?

### ISSUE THREAD

**Issue by aoussou** [korlibs/korau#31]
Hello,

I'm looking for a way of checking whether a sound is till playing in KMP (desktop only). It seems like the "playing" attributes of NativeSoundChannel might be what I'm looking for but I've been unable to access it.

Approach 1:

    val nsp = NativeSoundProvider()
    val wavBytes =  File(path).readBytes()
    sound = nsp.createSound(wavBytes)
    sound.playAndWait()

sound has no attribute "playing"

Approach 2:

        val sound = resourcesVfs[path].readSound()
        val channel = sound.play()
        channel.await()

This doesn't work because it can't find the file in "src/main/resources"

Is there an alternative to check whether the sound is playing?

**Comment by soywiz** [comment]
First, let's clarify one thing:

* `Sound` - Represents the waveform. It has not concept of `playing`. It has `length`, but also no `position`.
* `SoundChannel` - A waveform from a `Sound` that is playing. This has the concept of `position` and `playing` and inherits the `length` from the waveform.

With that said. `createSound` and `readSound` gives you a `Sound` instance, the non-playing waveform.
When you call the `Sound.play` method, you get a `SoundChannel` that is reproducing that waveform, in approach 2 you are calling the `.await`. The .await suspends the function until the `playing` ends being false. So `channel.playing` should be available, but if you access it after the `.await` it will be false.
If you plan to use `playAndWait`, you just know that the sound stopped being played after the `playAndWait` method. If you are launching it in parallel, you can trigger the code you need to be aware of that after the playAndWait or await calls.

If you are having problems loading a file, that might be a separate issue. If you have a local `path` instead of using resourcesVfs, you can use `currentLocalVfs[path]` (for current dir) or `localVfs(path)` (for any point in the local filesystem)

We use github for reporting bugs and feature requests, for questions we use a discord server here: https://discord.korge.org/

Related issue: https://github.com/korlibs/docs.korge.org/issues/19

There are two pages in the documentation about audio:
* https://docs.korge.org/korge/reference/audio/
* https://docs.korge.org/korau/
